tests/console/yast2_registration.pm

```diff
@@ -61,15 +61,16 @@ sub run {
 
     zypper_call "in yast2-registration";
 
-    cleanup_registration;
     my $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'registration');
-    assert_screen([qw(yast2_registration-overview yast2_registration-registration-page)]);
-
-    send_key "alt-e" if (match_has_tag "yast2_registration-overview");
+    assert_screen 'yast2_registration-overview';
+    send_key "alt-e";
+    assert_screen 'yast2_registration-registration-page-registered';
     register_system_and_add_extension;
     wait_serial("$module_name-0", 200) || die "'yast2 $module_name' didn't finish";
 
-    assert_script_run "SUSEConnect --status-text |grep -A3 -E 'SUSE Linux Enterprise Server|Web and Scripting Module' | grep -qE '^\\s+Registered'";
+    # Check via YaST if the system is already registered
+    $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'registration');
+    assert_screen 'yast2_registration-overview';
```

Comment:
Before this change the SUSEConnect check the register status of module such as web and scripting, but now ONLY check the yast register status? I feel loss some check points compare with old implementation.

Reply:
So I have tried checking registration state of individual modules and, although the idea seems good, checking the list of modules with needles can get flaky given the list of products that this is supposed to run on. For instance SLES 12-SP5 has a completely different module listing than SLES 15-SP4 as can be seen [here](https://openqa.suse.de/tests/9094062#step/yast2_registration/21).
So although we do lose this specific coverage compared to the old command line check, for the sake of stability and consistency of the test on all products, I would suggest that we do not check individual module registration with needles.

Reply:
Yes,  check modules status using yast needle will trigger even more trouble compare with command.
OK for me if you think loss coverage can be acceptable.

Reply:
let's see what is the less evil, if the needling in product in maintenance which should be sort of stable, or checking with command SUSEConnect when there are network glitches.